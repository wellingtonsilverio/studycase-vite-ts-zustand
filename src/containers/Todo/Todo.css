.Todo {
  margin-top: 6em;
}

.Todo .Input {
  width: 100%;

  display: flex;
  justify-content: center;
}
.Todo .Input input {
  flex: 1;

  margin-right: 0.5em;
  padding: 0 0.5em;

  font-size: 1.2em;
}

.Todo ul {
  padding-left: 0;
}

.Todo ul li {
  display: flex;
  align-items: center;
  justify-content: space-between;

  margin-bottom: 0.2em;
}

.Todo ul li a {
  margin: 0.5em;

  color: #fff;

  cursor: pointer;
}
