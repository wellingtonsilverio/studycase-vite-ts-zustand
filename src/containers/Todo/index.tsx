import Button from "../../components/Button";
import { useTodoState } from "../../hooks/store/todo";

import './Todo.css';

function getInputById(id: string) {
  return document.getElementById(id) as HTMLInputElement;
}
function clearInput(input: HTMLInputElement) {
  input.value = "";
}

const Todo: React.FC = () => {
  const todos = useTodoState(state => state.todos);
  const addTodo = useTodoState(state => state.addTodo);
  const removeTodo = useTodoState(state => state.removeTodo);

  const handleAddTodo = () => {
    const inputTitle = getInputById('title');
    
    addTodo(inputTitle?.value || "");

    clearInput(inputTitle);
  }

  const handleRemoveTodo = (key: number) => {
    removeTodo(key);
  }

  return (
    <div className="Todo">
      <h1>Lista de afazeres</h1>
      <div className="Input">
        <input id="title" /><Button onClick={handleAddTodo}>Adicionar</Button>
      </div>
      <ul>
        {todos?.map(todo => (
          <li key={todo.key}>{todo.label} <a onClick={() => handleRemoveTodo(todo.key)}>X</a></li>
        ))}
      </ul>
    </div>
  );
};

export default Todo;
