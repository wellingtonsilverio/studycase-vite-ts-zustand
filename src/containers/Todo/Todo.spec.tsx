import recoilRender from "../../utils/jest/recoilRender"
import Todo from './';

describe('Todo Container', () => {
  it('should render Todo', () => {
    const { container } = recoilRender(<Todo />);

    expect(container).toMatchSnapshot();
  });
});
