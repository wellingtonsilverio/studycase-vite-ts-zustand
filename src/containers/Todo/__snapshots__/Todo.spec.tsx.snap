// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Container should render Todo 1`] = `
<div>
  <div />
</div>
`;
